src/dumbo/src/tcp/handler.rs

```diff
@@ -148,6 +149,48 @@ enum RecvSegmentOutcome {
     UnexpectedSegment(bool),
 }
 
+/// build_response build micro_http::Response with given http version, status code, and body.
+///
+pub fn build_response(http_version: Version, status_code: StatusCode, body: Body) -> Response {
+    let mut response = Response::new(http_version, status_code);
+    response.set_body(body);
+    response
+}
+
+/// parse_request parses the request and build response by given callback function.
+///
+pub fn parse_request(request_bytes: &[u8], callback: fn(Request) -> Response) -> Response {
+    let request = Request::try_from(request_bytes);
+    match request {
+        Ok(request) => callback(request),
+        Err(e) => match e {
+            RequestError::InvalidHttpVersion(err_msg) => build_response(
+                Version::default(),
+                StatusCode::NotImplemented,
+                Body::new(err_msg.to_string()),
+            ),
+            RequestError::InvalidUri(err_msg) | RequestError::InvalidHttpMethod(err_msg) => {
+                build_response(
+                    Version::default(),
+                    StatusCode::BadRequest,
+                    Body::new(err_msg.to_string()),
+                )
+            }
+            RequestError::InvalidRequest => build_response(
+                Version::default(),
+                StatusCode::BadRequest,
+                Body::new("Invalid request.".to_string()),
+            ),
+            RequestError::InvalidHeader => build_response(
+                Version::default(),
+                StatusCode::BadRequest,
+                Body::new("Invalid headers.".to_string()),
+            ),
+            RequestError::UnsupportedHeader => unreachable!(),
```

Comment:
This will panic if the `Request` will contain an unsupported header. This can happen when trying to parse a `Request` and the `Content-type` is set to a content type different than `text/plain` or `application/json`. Can we return here a `BadRequest` error response as well?